### PR TITLE
emit_x64_vector: Vectorize 32-bit variants of paired min/max

### PR DESCRIPTION
**Pull request by lioncash**
Gets rid of the fallbacks for these cases.

### CODE REVIEW

File: src/backend_x64/emit_x64_vector.cpp [hunk 1]
```diff
@@ -1973,9 +2039,35 @@ void EmitX64::EmitVectorPairedMinU16(EmitContext& ctx, IR::Inst* inst) {
...
+        code.movdqa(y, code.MConst(xword, 0x8000000080000000, 0x8000000080000000));
+
+        const Xbyak::Xmm tmp2 = ctx.reg_alloc.ScratchXmm();
+        code.movdqa(tmp2, tmp1);
+
+        code.pxor(tmp2, y);
+        code.pxor(y, x);
+        code.pcmpgtd(y, tmp2);
+        code.pand(tmp1, y);
+        code.pand(y, x);
```
Comment (MerryMage) [inline src/backend_x64/emit_x64_vector.cpp]: this line likely should be
`code.pandn(y, x);`

Comment (lioncash) [reply src/backend_x64/emit_x64_vector.cpp]: oh, whoops, forgot to amend that into the commit

**Comment by MerryMage** [comment]
Force-pushed a patch:

```
diff --git a/src/backend_x64/emit_x64_vector.cpp b/src/backend_x64/emit_x64_vector.cpp
index 4d27665a..7629c99b 100644
--- a/src/backend_x64/emit_x64_vector.cpp
+++ b/src/backend_x64/emit_x64_vector.cpp
@@ -1972,17 +1972,18 @@ void EmitX64::EmitVectorPairedMaxU32(EmitContext& ctx, IR::Inst* inst) {

         ctx.reg_alloc.DefineValue(inst, x);
     } else {
-        code.movdqa(y, code.MConst(xword, 0x8000000080000000, 0x8000000080000000));
+        const Xbyak::Xmm tmp3 = ctx.reg_alloc.ScratchXmm();
+        code.movdqa(tmp3, code.MConst(xword, 0x8000000080000000, 0x8000000080000000));

         const Xbyak::Xmm tmp2 = ctx.reg_alloc.ScratchXmm();
         code.movdqa(tmp2, x);

-        code.pxor(tmp2, y);
-        code.pxor(y, tmp1);
-        code.pcmpgtd(y, tmp2);
-        code.pand(tmp1, y);
-        code.pandn(y, x);
-        code.por(tmp1, y);
+        code.pxor(tmp2, tmp3);
+        code.pxor(tmp3, tmp1);
+        code.pcmpgtd(tmp3, tmp2);
+        code.pand(tmp1, tmp3);
+        code.pandn(tmp3, x);
+        code.por(tmp1, tmp3);

         ctx.reg_alloc.DefineValue(inst, tmp1);
     }
@@ -2054,17 +2055,18 @@ void EmitX64::EmitVectorPairedMinU32(EmitContext& ctx, IR::Inst* inst) {

         ctx.reg_alloc.DefineValue(inst, x);
     } else {
-        code.movdqa(y, code.MConst(xword, 0x8000000080000000, 0x8000000080000000));
+        const Xbyak::Xmm tmp3 = ctx.reg_alloc.ScratchXmm();
+        code.movdqa(tmp3, code.MConst(xword, 0x8000000080000000, 0x8000000080000000));

         const Xbyak::Xmm tmp2 = ctx.reg_alloc.ScratchXmm();
         code.movdqa(tmp2, tmp1);

-        code.pxor(tmp2, y);
-        code.pxor(y, x);
-        code.pcmpgtd(y, tmp2);
-        code.pand(tmp1, y);
-        code.pandn(y, x);
-        code.por(tmp1, y);
+        code.pxor(tmp2, tmp3);
+        code.pxor(tmp3, x);
+        code.pcmpgtd(tmp3, tmp2);
+        code.pand(tmp1, tmp3);
+        code.pandn(tmp3, x);
+        code.por(tmp1, tmp3);

         ctx.reg_alloc.DefineValue(inst, tmp1);
     }
```

`y` was not a scratchable register.

**Comment by lioncash** [comment]
could easily just make `y` a scratchable reg and not need to introduce another tmp, as far as I can tell.

**Pull request merged**